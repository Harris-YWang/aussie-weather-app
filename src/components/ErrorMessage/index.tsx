import React from 'react';
import styled from 'styled-components';

const ErrorTitle = styled.div`
  display: block;
  font-size: 1.17em;
  margin-block-start: 1em;
  margin-block-end: 1em;
  margin-inline-start: 0px;
  margin-inline-end: 0px;
  font-weight: bold;
`;

const ErrorText = styled.div`
  display: block;
  margin-block-start: 1em;
  margin-block-end: 1em;
  margin-inline-start: 0px;
  margin-inline-end: 0px;
`;

const ErrorMessage = (props: { keyword: string }) => {
  return (
    <div>
      <ErrorTitle>Sorry, No results</ErrorTitle>
      <ErrorText>Nothing came up for that "{props.keyword}" search</ErrorText>
    </div>
  );
};

export default ErrorMessage;
