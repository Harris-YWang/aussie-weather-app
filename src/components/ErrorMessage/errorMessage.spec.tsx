import React from 'react';
import { shallow } from 'enzyme';
import ErrorMessage from './index';

describe('ErrorMessage Component', () => {
  it('should render without crashing', () => {
    const keyword = 'keyword';
    const wrapper = shallow(<ErrorMessage keyword={keyword} />);

    expect(wrapper).toMatchSnapshot();
  });
});
