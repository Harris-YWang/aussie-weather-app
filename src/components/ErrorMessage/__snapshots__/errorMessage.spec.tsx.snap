// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage /> renders without crashing 1`] = `ShallowWrapper {}`;

exports[`ErrorMessage Component renders without crashing 1`] = `ShallowWrapper {}`;

exports[`ErrorMessage Component should render without crashing 1`] = `ShallowWrapper {}`;
